File: ecopoints/app/src/main/java/at/htl/ecopoints/db/services/UserService.java
package at.htl.ecopoints.db.services;
import java.util.List;

import at.htl.ecopoints.model.User;
import okhttp3.Response;

public class UserService extends Service {
    private final String endPoint = "user";

    public Response createUser(User user) throws Exception {
        return super.create(user, endPoint);
    }

    public Response updateUser(User user, Long id) throws Exception {
        return super.update(user, endPoint, id.toString());
    }

    public Response deleteUser(Long id) throws Exception {
        return super.delete(endPoint, id.toString());
    }

    public User getUserById(Long id) throws Exception {
        return super.getById(endPoint, id.toString());
    }

    public List<User> getAllUsers() throws Exception {
        return super.getAll(endPoint);
    }
}

File: ecopoints/app/src/main/java/at/htl/ecopoints/backendService/UserService.kt
package at.htl.ecopoints.backendService

import at.htl.ecopoints.model.User
import okhttp3.Response

class UserService: Service() {
    private final val endPoint = "user"

    fun createUser(user: User): Response {
        return super.create(user, endPoint)
    }

    fun updateUser(user: User, id: Long): Response {
        return super.update(user, endPoint, id.toString())
    }

    fun deleteUser(id: Long): Response {
        return super.delete(endPoint, id.toString())
    }

    fun getUserById(id: Long): User? {
        return super.getById<User>(endPoint, id.toString())
    }

    fun getAllUsers(): List<User>? {
        return super.getAll<User>(endPoint)
    }
}